File: scala/src/Benchmark.scala
/**
  * @author Artur Bosch
  */
object Benchmark {

  def main(args: Array[String]): Unit = {
    val time: Long = benchmark(() => LinesOfCode.linesOfCode("./scala/src/LinesOfCode.scala"))
    println(s"Time: $time ms")
  }

  def benchmark(block: () => Unit): Long = {
    (1 to 10).map(n => measure(block)).sum / 10
  }

  def measure(block: () => Unit): Long = {
    val start: Long = System.currentTimeMillis()
    block()
    System.currentTimeMillis() - start
  }
}

File: kotlin/src/Benchmark.kt
import kotlin.system.measureTimeMillis

/**
 * @author Artur Bosch
 */

fun main(args: Array<String>) {
	val time = benchmark { linesOfCode("./kotlin/src/LinesOfCode.kt") }
	println("Time: $time ms")
}

fun benchmark(block: () -> Unit): Long {
	return (1..10).map {
		measureTimeMillis {
			block()
		}
	}.sum() / 10
}